### PR TITLE
fix(cirrus): don't require context

### LINKED ISSUE #13105: Allow cirrus to take empty context
Cirrus is not configured to take empty context, it returns 400, we should allow Cirrus to take empty context as the input

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-5748)

### PR DESCRIPTION
**Pull request by relud**
Because

- context should be optional

This commit

- changes context from required to using a default value of `{}`

Fixes #13105

### CODE REVIEW

**Review by jaredlockhart — COMMENTED** [review]
Oh I'm not sure we should make the context completely optional entirely, I only meant that it could be empty.  I dunno is there a big difference between the two cases? 🤔 Should we keep it in but support an empty object?  I worry about making it optional entirely and some consumers baking that assumption into their code and then later we find some other reason to depend on the object in some case?

**Comment by relud** [comment]
> Oh I'm not sure we should make the context completely optional entirely, I only meant that it could be empty. I dunno is there a big difference between the two cases?

I don't think so

> 🤔 Should we keep it in but support an empty object?

That is a trivial change to make

> I worry about making it optional entirely and some consumers baking that assumption into their code and then later we find some other reason to depend on the object in some case?

i don't think so, but i don't feel strongly

File: cirrus/server/tests/test_main.py [hunk 1]
```diff
@@ -122,23 +122,6 @@ def test_get_features_v2_with_required_field(client):
...
-                {
-                    "type": "missing",
-                    "loc": ["body", "context"],
-                    "msg": "Field required",
-                    "input": {"client_id": "4a1d71ab-29a2-4c5f-9e1d-9d9df2e6e449"},
-                }
-            ],
-        ),
         (
             {},
```
Comment (yashikakhurana) [inline cirrus/server/tests/test_main.py]: I will not remove this test case and instead this should return valid default feature now

File: cirrus/server/tests/test_main.py [hunk 2]
```diff
@@ -149,12 +132,6 @@ def test_get_features_v2_with_required_field(client):
                     "msg": "Field required",
                     "input": {},
                 },
-                {
-                    "type": "missing",
-                    "loc": ["body", "context"],
-                    "msg": "Field required",
-                    "input": {},
-                },
```
Comment (yashikakhurana) [inline cirrus/server/tests/test_main.py]: same here

File: cirrus/server/tests/test_main.py [hunk 3]
```diff
@@ -444,13 +392,7 @@ def test_get_features_v1_with_nimbus_preview_flag(client):
...
 
 def test_get_features_v2_with_nimbus_preview(client):
-    request_data = {
-        "client_id": "4a1d71ab-29a2-4c5f-9e1d-9d9df2e6e449",
-        "context": {
-            "key1": "value1",
-            "key2": {"key2.1": "value2", "key2.2": "value3"},
-        },
-    }
+    request_data = {"client_id": "4a1d71ab-29a2-4c5f-9e1d-9d9df2e6e449"}
```
Comment (yashikakhurana) [inline cirrus/server/tests/test_main.py:472]: split this into two one with context with key value and one with `{}`

File: cirrus/server/tests/test_main.py (hunk 1 above)
Comment (relud) [reply cirrus/server/tests/test_main.py]: these tests are called like `test_get_features_v1_missing_required_field`, so I think it's misleading for future developers to have cases on it that aren't missing a required field.

Instead I modified `test_get_features_v1_with_nimbus_preview_flag` and `test_get_features_v2_with_nimbus_preview` to cover testing empty context (and updated to parametrize for both empty and nonempty per your review).

File: cirrus/server/tests/test_main.py (hunk 2 above)
Comment (relud) [reply cirrus/server/tests/test_main.py]: this is no longer touched, so it's no longer relevant, but also this isn't a test case that was removed, it was expecting a different return value

File: cirrus/server/tests/test_main.py [hunk 4]
```diff
@@ -99,8 +99,9 @@ def test_get_features_v2_with_required_field(client):
             },
             status.HTTP_400_BAD_REQUEST,
             "Client ID value is missing or empty",
+            id="empty_client_id",
```
Comment (relud) [inline cirrus/server/tests/test_main.py:102]: I went through all the parametrized tests in this file and added ids because i couldn't tell them apart when debugging

File: cirrus/server/tests/test_main.py [hunk 5]
```diff
@@ -443,13 +451,23 @@ def test_get_features_v1_with_nimbus_preview_flag(client):
     }
 
 
-def test_get_features_v2_with_nimbus_preview(client):
+@pytest.mark.parametrize(
```
Comment (yashikakhurana) [inline cirrus/server/tests/test_main.py:454]: oh thank you, yeah perfect, v1 and v2 both with nimbus preview flag, can you also do the same for without preview flag

**Review by yashikakhurana — APPROVED** [review]
Perfect, all in one place, thank you for adding those params 🙏
